flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/rest/SessionCaseITTest.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.flink.table.gateway.rest.message.session.OpenSessionRequestBody;
+import org.apache.flink.table.gateway.rest.message.session.OpenSessionResponseBody;
+import org.apache.flink.table.gateway.rest.message.session.SessionMessageParameters;
+import org.apache.flink.table.gateway.service.session.Session;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static org.apache.flink.table.gateway.rest.handler.session.CloseSessionHandler.CLOSE_MESSAGE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+/**
+ * Test basic logic of handlers inherited from {@link AbstractSqlGatewayRestHandler} in session
+ * related cases.
+ */
+class SessionCaseITTest extends RestAPIITTestBase {
+
+    private static final String SESSION_NAME = "test";
+    private static final Map<String, String> properties = new HashMap<>();
+    private static final int SESSION_NUMBER = 10;
+
+    static {
+        properties.put("k1", "v1");
+        properties.put("k2", "v2");
+    }
+
+    OpenSessionHeaders openSessionHeaders = OpenSessionHeaders.getInstance();
+    OpenSessionRequestBody openSessionRequestBody =
+            new OpenSessionRequestBody(SESSION_NAME, properties);
+    EmptyMessageParameters emptyParameters = EmptyMessageParameters.getInstance();
+
+    CloseSessionHeaders closeSessionHeaders = CloseSessionHeaders.getInstance();
+    EmptyRequestBody emptyRequestBody = EmptyRequestBody.getInstance();
+    SessionMessageParameters sessionMessageParameters;
+
+    @Test
+    void testWhenCreateMultiSessions() throws Exception {
+        Set<String> sessionHandleIds = new HashSet<>();
+        for (int num = 0; num < SESSION_NUMBER; ++num) {
+            CompletableFuture<OpenSessionResponseBody> response =
+                    restClientUtils.sendRequest(
+                            openSessionHeaders, emptyParameters, openSessionRequestBody);
+            String sessionHandleId = response.get().getSessionHandle();
+            sessionHandleIds.add(sessionHandleId);
+            assertThat(sessionHandleId).isNotNull();
+            assertThat(
+                            SQL_GATEWAY_SERVICE_EXTENSION
+                                    .getSessionManager()
+                                    .getSession(
+                                            new SessionHandle(UUID.fromString(sessionHandleId))))
+                    .isNotNull();
+        }
+        assertThat(sessionHandleIds).hasSize(SESSION_NUMBER);
+    }
+
+    @Test
+    void testWhenCreateAndCloseSessions() throws Exception {
```

Comment:
This case completely covers `testWhenCreateMultiSessions`.

Reply:
Has been removed.